keanu-project/src/main/java/io/improbable/keanu/algorithms/variational/optimizer/Variable.java

```diff
@@ -1,10 +1,19 @@
 package io.improbable.keanu.algorithms.variational.optimizer;
 
-public interface Variable<T> {
+import io.improbable.keanu.network.VariableState;
+import io.improbable.keanu.vertices.Samplable;
+
+public interface Variable<VALUE, STATE extends VariableState> extends Samplable<VALUE> {
```

Comment:
Note that this now extends `Samplable`. The `sample` method is only called in one place: `PriorProposalDistribution#setFor`.

We could probably eliminate it, although I think conceptually it is valid for a `Variable` to be `Samplable`.